gammapy/irf/energy_dispersion.py

```diff
@@ -541,6 +545,291 @@ def gauss_energy_dispersion_matrix(ebounds, sigma=0.2):
+        header = hdu.header
+        e_lo = EnergyBounds(data['ETRUE_LO'].squeeze(), header['TUNIT1'])
+        e_hi = EnergyBounds(data['ETRUE_HI'].squeeze(), header['TUNIT2'])
+        o_lo = Angle(data['THETA_LO'].squeeze(), header['TUNIT5'])
+        o_hi = Angle(data['THETA_HI'].squeeze(), header['TUNIT6'])
+        m_lo = data['MIGRA_LO'].squeeze()
+        m_hi = data['MIGRA_HI'].squeeze()
+        matrix = data['MATRIX'].squeeze()
+
+        return cls(e_lo, e_hi, m_lo, m_hi, o_lo, o_hi, matrix)
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyDispersion2D` from ``GCTAEdisp2D`` format FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdulist = fits.open(filename)
+        return cls.from_fits(hdulist['ENERGY DISPERSION'])
+
+    def evaluate(self, offset=None, e_true=None, migra=None):
+        """Probability for a given offset, true energy, and migration
+        """
+
+        if offset is None:
+            offset = self.offset
+        if e_true is None:
+            e_true = self.energy
+        if migra is None:
+            migra = self.migra
+
+        if not isinstance(e_true, Quantity):
+            raise ValueError("Energy must be a Quantity object.")
+        if not isinstance(offset, Angle):
+            raise ValueError("Offset must be an Angle object.")
+
+        offset = offset.to('degree')
+        e_true = e_true.to('TeV')
+
+        val = self._eval(offset=offset, e_true=e_true, migra=migra)
+
+        return val
+
+    def _eval(self, offset=None, e_true=None, migra=None):
+
+        x = np.asarray(offset.value)
+        y = np.asarray(migra)
+        z = np.asarray(np.log10(e_true.value))
+        ax = [x, y, z]
+
+        in_shape = (ax[0].size, ax[1].size, ax[2].size)
+
+        for i, s in enumerate(ax):
+            if ax[i].shape == ():
+                ax[i] = ax[i].reshape(1)
+
+        # There is a bug here that could be investigated
+        # When energy[0] is given to the interpolator is out if bounds
+        # This does not happen when an array (e.g. energy[0:2]) is given
+
+        pts = [[xx, yy, zz] for xx in ax[0] for yy in ax[1] for zz in ax[2]]
+
+        val_array = self._linear(pts)
+        return val_array.reshape(in_shape).squeeze()
+
+    def to_energy_dispersion(self, e_true=None, e_reco=None, offset=None):
+        """Detector response R(Delta E_reco, Delta E_true)
+
+        Probability to reconstruct an energy in a given true energy band
+        in a given reconstructed energy band
+
+        Parameters
+        ----------
+        e_true : `~gammapy.spectrum.energy.EnergyBounds`, None
+            True energy axis
+        e_reco : `~gammapy.spectrum.energy.EnergyBounds`
+            Reconstructed energy axis
+        offset : `~astropy.coordonates.Angle`
+            Offset
+
+        Returns
+        -------
+        edisp : `~gammapy.irf.energy_dispersion.EnergyDispersion`
```

Comment:
This should be `~gammapy.irf.EnergyDispersion`

I don't think `~gammapy.irf.energy_dispersion.EnergyDispersion` will resolve to a link in the HTML docs.

Read [this](https://gammapy.readthedocs.org/en/latest/development/howto.html#where-should-i-import-from) and then check your HTML docs locally via
```
python setup.py build_sphinx
open docs/_build/html/index.html
```